### DISCUSSION TITLE
Currency conversion

### DISCUSSION THREAD

**Discussion by Fred-Vatin** [hoothin/SearchJumper#180] [Search engines]
This updates the broken default search engines to convert currencies.

First go to https://apilayer.com and create an account. Then go to https://api.apilayer.com/fixer/convert and subscribe. Copy your api key. There are other API providers [here](https://apilayer.com/marketplace/category/currency) you can subscribe. You’ll just need to change the URL in the search engines to match them.

Now you can import those search engines and edit them to your taste.

```
{
  "name": "💲USD to EUR",
  "url": "showTips:https://api.apilayer.com/fixer/convert?to=EUR&from=USD&amount=1&apikey=%template{apilayer key} \n\n{name}\n<br/>\n<i>%s USD = {json.result|*%s.replace(/\\D/g,'')} EUR</i>",
  "kwFilter": "\\s*\\d+\\s*"
}
```
```
{
  "name": "💲Currency convert",
  "url": "showTips:https://api.apilayer.com/fixer/convert?from=%input{From currency/US Dollar/Euro/Japanese Yen/China Yuan,USD/EUR/JPY/CNY}&to=%input{Target currency/US Dollar/Euro/Japanese Yen/China Yuan,USD/EUR/JPY/CNY}&amount=1&apikey=%template{apilayer key}\n\n{name}\n<br/><i>%sr {json.query.from} = {json.result|*%sr.replace(/\\D/,'')} {json.query.to}</i>",
  "kwFilter": "\\s*\\d+\\s*"
}
```

**Comment by neoOpus** [comment]
https://fixer.io/
